client/src/main.rs

```diff
@@ -98,7 +98,13 @@ async fn run(
 	)
 	.await?;
 
-	spawn_in_span(shutdown.with_cancel(p2p_event_loop.run(ot_metrics.clone())));
+	spawn_in_span(shutdown.with_cancel(p2p_event_loop.run()));
+
+	let metrics_clone = ot_metrics.clone();
+	tokio::spawn(async move {
+		let stream = BroadcastStream::new(event_receiver.resubscribe());
+		metrics_clone.handle_event_stream(stream).await
+	});
```

Comment:
We should run event loop after the call to resubscribe to avoid skipping an event.